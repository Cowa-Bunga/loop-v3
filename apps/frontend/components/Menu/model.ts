export const menuModel = [
  { id: 1, icon: '', title: 'Dashboard', link: '/' },
  { id: 2, icon: '', title: 'All Orders', link: '/orders' },
  { id: 3, icon: '', title: 'Driver Details', link: '/' },
  { id: 4, icon: '', title: 'Import Orders', link: '/' },
  { id: 5, icon: '', title: 'Training Centre', link: '/' },
  { id: 6, icon: '', title: 'Support', link: '/' }
]

export const menuBottomModel = [
  { id: 7, icon: '', title: 'Client Admin', link: '/' },
  { id: 8, icon: '', title: 'Fleet Management', link: '/' },
  { id: 9, icon: '', title: 'Log Out', action: () => 'signOut' }
]
