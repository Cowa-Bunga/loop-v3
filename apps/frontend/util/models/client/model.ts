export const MODEL_CLIENT_EMPTY: IClient = {
  client_id: '',
  name: '',
  user_id: '',
  permissions_id: '',
  logo: '',
  permissions: {
    scopes: [],
    fleet: false,
    administrator: false
  }
}
