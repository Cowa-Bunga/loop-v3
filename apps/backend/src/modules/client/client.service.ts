import { Injectable } from '@nestjs/common'
import * as admin from 'firebase-admin'
import { DocumentSnapshot } from 'firebase-admin/firestore'

@Injectable()
export class ClientService {
  async getClientById(client_id: string): Promise<DocumentSnapshot> {
    const db = admin.firestore()
    const client = await db.collection('clients').doc(client_id).get()

    return client
  }

  async getClientByKey(apiKey: string): Promise<DocumentSnapshot> {
    const db = admin.firestore()
    const clients = await db.collection('clients').where('api.key', '==', apiKey).limit(1).get()
    return clients.docs.pop()
  }
}
