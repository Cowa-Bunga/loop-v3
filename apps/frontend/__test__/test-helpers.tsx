import { fireEvent, render, screen, within } from '@testing-library/react'
import { FirebaseAppProvider } from 'reactfire'
import { firebaseConfig } from '@util/lib/firebase'
import '@locale/config'

if (process.env.NODE_ENV === 'test') {
  console.info = console.log = console.warn = console.error = () => ''
}

export const renderWithProviders = (ui: JSX.Element) => {
  return {
    ...render(ui, {
      wrapper: ({ children }) => (
        <FirebaseAppProvider firebaseConfig={firebaseConfig}>
          {children}
        </FirebaseAppProvider>
      )
    })
  }
}

//TODO: workout why this cannot click on the first item in the select list
export const selectMaterialUiSelectOption = (element, optionText) => {
  // Select list is actually a button which we need to trigger to get
  // the listbox to appear with all the select "options
  const button = within(element).getByRole('button')
  fireEvent.mouseDown(button)

  // Get the listbox and all the options
  const listbox = within(screen.getByRole('presentation')).getByRole('listbox')
  const option = within(listbox).queryByText(optionText)
  // click on the option with the text we want
  fireEvent.click(option)
}
