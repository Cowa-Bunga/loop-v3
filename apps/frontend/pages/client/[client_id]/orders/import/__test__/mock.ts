export const rows = [
  {
    id: 1,
    col1: 'today',
    col2: '2',
    col3: '1',
    col4: '1',
    col5: 'MAsood',
  },
  {
    id: 2,
    col1: 'tomorrow',
    col2: '20',
    col3: '10',
    col4: '10',
    col5: 'Tony Stank',
  },
  {
    id: 3,
    col1: 'together',
    col2: '90',
    col3: '90',
    col4: '0',
    col5: 'Monkey D Luffy',
  }
]

export const columns = [
  { field: 'col1', headerName: 'Date', minWidth: 288 },
  { field: 'col2', headerName: 'Total Orders', minWidth: 288 },
  { field: 'col3', headerName: 'Succeeded', minWidth: 288 },
  { field: 'col4', headerName: 'Failed', minWidth: 288 },
  { field: 'col5', headerName: 'Imported By', minWidth: 288 },
  { field: 'actions', headerName: '', minWidth: 200 }
]
