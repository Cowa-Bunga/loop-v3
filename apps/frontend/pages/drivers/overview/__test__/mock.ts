export const rows = [
  {
    id: 1,
    col1: 'ORDER001111',
    col2: 'John Smith',
    col3: '23/3/2023 14:21:33',
    col4: '23/3/2023 14:23:12',
    col5: 'CPT 0034',
    col6: 'delivered',
    col7: 3,
    col8: 'ON DEMAND'
  },
  {
    id: 2,
    col1: 'ORDER001211',
    col2: 'Sally Rodger',
    col3: '23/3/2023 14:21:33',
    col4: '23/3/2023 14:23:12',
    col5: 'CPT 0034',
    col6: 'delivered',
    col7: 3,
    col8: 'ON DEMAND'
  },
  {
    id: 3,
    col1: 'ORDER001113',
    col2: 'Robert Pixies',
    col3: '23/3/2023 14:21:33',
    col4: '23/3/2023 14:23:12',
    col5: 'CPT 0034',
    col6: 'delivered',
    col7: 3,
    col8: 'ON DEMAND'
  }
]

export const columns = [
  { field: 'col1', headerName: 'DRIVER', minWidth: 180 },
  { field: 'col2', headerName: 'ASSIGNED TRIP', minWidth: 180 },
  { field: 'col3', headerName: 'LAST ONLINE', minWidth: 180 },
  { field: 'col4', headerName: 'AVERAGE TIME', minWidth: 180 },
  { field: 'col5', headerName: 'EXCEPTIONS', minWidth: 180 },
  { field: 'col6', headerName: 'STATUS', minWidth: 180 },
  { field: 'col7', headerName: 'RATING', minWidth: 180 },
  { field: 'col8', headerName: 'TYPE', minWidth: 180 },
  { field: 'actions', headerName: '', minWidth: 200 }
]
