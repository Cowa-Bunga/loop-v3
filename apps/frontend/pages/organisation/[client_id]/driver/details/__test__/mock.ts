export const rows = [
  {
    id: 1,
    name: 'Frikkie Spizo',
    code: '2345676543',
    email: 'frikkz@spizo.co',
    type: 'car',
    client: 'Sibas deliveries',
    hubs: 'Cape Town Hub',
    status: 'Active',
    trips: ''
  }
]

export const columns = [
  { field: 'name', headerName: 'DRIVER NAME', minWidth: 160 },
  { field: 'code', headerName: 'CODE', minWidth: 120 },
  { field: 'email', headerName: 'EMAIL ADDRESS', minWidth: 140 },
  { field: 'type', headerName: 'VEHICLE TYPE', minWidth: 120 },
  { field: 'client', headerName: 'ASSOCIATED CLIENT', minWidth: 180 },
  { field: 'hubs', headerName: 'ASSOCIATED HUBS', minWidth: 180 },
  { field: 'status', headerName: 'STATUS', minWidth: 120 },
  { field: 'trips', headerName: 'MANAGE TRIPS', minWidth: 160 }
]
