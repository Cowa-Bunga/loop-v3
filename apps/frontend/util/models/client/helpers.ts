import { defaultsDeep } from 'lodash'
import { MODEL_CLIENT_EMPTY } from './model'

export const modelClientList = (
  data: { [s: string]: unknown } | ArrayLike<unknown>
): IClient[] => {
  return Object.entries(data)
    .map(modelClientData)
    .map((el: never) => el['1'])
    .sort((a: IClient, b: IClient) => a.name.localeCompare(b.name))
}

export const modelClientData = (data: unknown): IClient => {
  return defaultsDeep(data, MODEL_CLIENT_EMPTY)
}
