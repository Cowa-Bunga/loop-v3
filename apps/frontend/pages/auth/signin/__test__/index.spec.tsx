/* eslint-disable @typescript-eslint/ban-ts-comment */

import React from 'react'
import SignIn from '../index.page'
// @ts-ignore
import { withMockAuth } from '@tomfreudenberg/next-auth-mock/jest'
import { renderWithProviders } from '../../../../__test__/test-helpers'
import { MODEL_CLIENT_EMPTY } from '@util/models/client/model'
import { fireEvent } from '@testing-library/react'

const mockRouter = jest.fn()
const mockUseSession = jest.fn()
const mockAuthFirebase = jest.fn()
const mockGetUser = jest.fn()
const mockRouterPush = jest.fn()
const mockActionSubmit = jest.fn()
const mockActionChange = jest.fn()
const mockUseMergeState = jest.fn()
jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: jest.fn().mockImplementation(() => mockRouter())
}))

jest.mock('@hooks', () => ({
  ...jest.requireActual('@hooks'),
  useSession: jest.fn().mockImplementation(() => mockUseSession()),
  useMergeState: jest.fn().mockImplementation(() => mockUseMergeState())
}))

jest.mock('@util/lib/firebase', () => ({
  ...jest.requireActual('@util/lib/firebase'),
  authFirebase: jest.fn().mockImplementation(() => mockAuthFirebase())
}))

// mock default export form ./Actions
jest.mock('../actions', () => ({
  default: jest.fn().mockImplementation(() => ({
    getUser: jest.fn().mockImplementation(() => mockGetUser()),
    submit: jest.fn().mockImplementation(() => mockActionSubmit()),
    change: jest.fn().mockImplementation(() => mockActionChange())
  })),
  __esModule: true
}))

describe('SignIn', () => {
  beforeEach(() => {
    mockUseSession.mockReturnValue({
      status: 'loading',
      data: {}
    })
    mockRouter.mockReturnValue({
      query: {}
    })
    mockAuthFirebase.mockResolvedValue({})
    mockGetUser.mockResolvedValue({})
    mockUseMergeState.mockReturnValue([
      {
        email: '',
        password: ''
      },
      jest.fn()
    ])
  })

  it('should render successfully', () => {
    const { baseElement } = renderWithProviders(withMockAuth(<SignIn />))

    expect(baseElement).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })

  it('should render successfully - error', () => {
    mockRouter.mockReturnValue({
      query: {
        error: 'error'
      }
    })

    const { baseElement } = renderWithProviders(withMockAuth(<SignIn />))

    expect(baseElement).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })

  it('should redirect the user to the map page', () => {
    mockUseSession.mockReturnValue({
      status: 'authenticated',
      data: {
        user: {
          clients: [MODEL_CLIENT_EMPTY]
        }
      }
    })
    mockRouter.mockReturnValue({
      query: {},
      push: mockRouterPush.mockResolvedValue({})
    })

    renderWithProviders(withMockAuth(<SignIn />))

    expect(mockRouterPush).toHaveBeenCalledWith('/map')
  })

  it('should redirect the user to the client select page', () => {
    mockUseSession.mockReturnValue({
      status: 'authenticated',
      data: {
        user: {
          clients: [MODEL_CLIENT_EMPTY, MODEL_CLIENT_EMPTY, MODEL_CLIENT_EMPTY]
        }
      }
    })
    mockRouter.mockReturnValue({
      query: {},
      push: mockRouterPush.mockResolvedValue({})
    })

    renderWithProviders(withMockAuth(<SignIn />))

    expect(mockRouterPush).toHaveBeenCalledWith('/auth/client_select')
  })

  it('should complete the login form', () => {
    const { baseElement } = renderWithProviders(withMockAuth(<SignIn />))

    const emailInput = baseElement.querySelector('#email')
    const passwordInput = baseElement.querySelector('#password')

    fireEvent.change(emailInput, { target: { value: 'test@test.com' } })
    fireEvent.change(passwordInput, { target: { value: 'password' } })

    expect(mockActionChange).toHaveBeenCalled()
  })

  it('should submit the login form', () => {
    mockUseMergeState.mockReturnValue([
      {
        email: 'test@test.com',
        password: 'password'
      },
      jest.fn()
    ])

    const { baseElement } = renderWithProviders(withMockAuth(<SignIn />))

    const submitButton = baseElement.querySelector('button[type="submit"]')

    fireEvent.click(submitButton)

    expect(mockActionSubmit).toHaveBeenCalled()
  })
})
