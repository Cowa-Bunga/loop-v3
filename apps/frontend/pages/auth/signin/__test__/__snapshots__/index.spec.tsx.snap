// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render successfully - error 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-tyohzp-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-1qvppxn-MuiTypography-root"
            >
              <img
                alt="logo"
                src="https://www.loop.co.za/wp-content/uploads/2021/12/Logo.svg"
                style="margin-top: 5px; height: 20px;"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LockIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
      <div
        class="MuiBox-root css-13yz848"
      >
        <div
          class="MuiBox-root css-1u8fws6"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
            role="alert"
          >
            <div
              class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="SuccessOutlinedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
                />
              </svg>
            </div>
            <div
              class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
            >
              Could not login. Please check your e-mail or password.
            </div>
          </div>
          <div
            style="height: calc(100vh - 100px); background-image: url(/login_background.png); background-size: cover;"
          >
            <main
              class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3k0qj7-MuiPaper-root-MuiCard-root"
              >
                <img
                  alt="logo"
                  src="/loop-log-icon.png"
                  style="height: 120px;"
                />
                <h1
                  class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                >
                  Sign In
                </h1>
                <form
                  class="MuiBox-root css-178yklu"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="email"
                          id="email-label"
                        >
                          Email Address
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="email"
                            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                            id="email"
                            name="email"
                            required=""
                            type="email"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-yjsfm1"
                            >
                              <span>
                                Email Address
                                 
                                *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="password"
                          id="password-label"
                        >
                          Password
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="new-password"
                            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                            id="password"
                            name="password"
                            required=""
                            type="password"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-yjsfm1"
                            >
                              <span>
                                Password
                                 
                                *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <br />
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth css-1ij77gt-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    Sign In
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
            </main>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`SignIn should render successfully 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div
        class="MuiBox-root css-i9gxme"
      >
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-tyohzp-MuiPaper-root-MuiAppBar-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
          >
            <div
              class="MuiTypography-root MuiTypography-h6 css-1qvppxn-MuiTypography-root"
            >
              <img
                alt="logo"
                src="https://www.loop.co.za/wp-content/uploads/2021/12/Logo.svg"
                style="margin-top: 5px; height: 20px;"
              />
            </div>
            <button
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="LockIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </header>
      </div>
      <div
        class="MuiBox-root css-13yz848"
      >
        <div
          class="MuiBox-root css-1u8fws6"
        >
          <div
            style="height: calc(100vh - 100px); background-image: url(/login_background.png); background-size: cover;"
          >
            <main
              class="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3k0qj7-MuiPaper-root-MuiCard-root"
              >
                <img
                  alt="logo"
                  src="/loop-log-icon.png"
                  style="height: 120px;"
                />
                <h1
                  class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
                >
                  Sign In
                </h1>
                <form
                  class="MuiBox-root css-178yklu"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="email"
                          id="email-label"
                        >
                          Email Address
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="email"
                            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                            id="email"
                            name="email"
                            required=""
                            type="email"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-yjsfm1"
                            >
                              <span>
                                Email Address
                                 
                                *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                          data-shrink="false"
                          for="password"
                          id="password-label"
                        >
                          Password
                          <span
                            aria-hidden="true"
                            class="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                          >
                             
                            *
                          </span>
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                        >
                          <input
                            aria-invalid="false"
                            autocomplete="new-password"
                            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                            id="password"
                            name="password"
                            required=""
                            type="password"
                            value=""
                          />
                          <fieldset
                            aria-hidden="true"
                            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                          >
                            <legend
                              class="css-yjsfm1"
                            >
                              <span>
                                Password
                                 
                                *
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <br />
                  <hr
                    class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                  />
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeLarge MuiButton-outlinedSizeLarge MuiButton-fullWidth css-1ij77gt-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    Sign In
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </form>
              </div>
            </main>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
